templates/commands/render.go

```diff
@@ -454,11 +454,38 @@ func (c *RenderCommand) checkInputsMissing(spec *model.Spec) []string {
 func executeSpec(ctx context.Context, spec *model.Spec, sp *stepParams) error {
 	logger := logging.FromContext(ctx).Named("executeSpec")
 
-	for _, step := range spec.Steps {
+	for i, step := range spec.Steps {
 		if err := executeOneStep(ctx, step, sp); err != nil {
 			return err
 		}
 		logger.Debugw("completed template action", "action", step.Action.Val)
+		if sp.flags.DebugScratchContents {
+			sb := &strings.Builder{}
+			fmt.Fprintf(sb, "Scratch dir contents after step %d (starting from 0), which is action type %q, defined at spec file line %d:\n",
```

Comment:
Should this be structured (K=V) instead of string interpolated?

Reply:
I just coded that up to see how readable it is, and it's much harder to read.

Here's what it loiks like as a plain string:

```
INFO	logScratchContents	commands/render.go:501	Scratch dir contents after step 1 (starting from 0), which is action type "string_replace", defined at spec file line 41:
  .editorconfig
  .eslintignore
  ... snip ...
```

And here's what it looks like with structured logging:
```
INFO	logScratchContents	commands/render.go:498	Scratch dir contents:   .editorconfig
  .eslintignore
...snip...
  public/index.html
  public/manifest.json
	{"step_index_starting_from_zero": 1, "action": "string_replace", "spec_yaml_line_number": 41}
```

I can change it if you really think it's important, but I think in this case it makes sense to choose readability.